src/runtime/manifest-primitive-parser/test/manifest-primitive-parser-test.ts

```diff
@@ -0,0 +1,228 @@
+    const manifest = await ManifestPrimitiveParser.parse(`
+    particle TestParticle in 'a.js'
+      list: reads [Product {}]
+      person: writes Person {}
+      thing: consumes
+        otherThing: provides
+    `);
+    const particles = ManifestPrimitiveParser.extract('particle', manifest);
+    assert.lengthOf(particles, 1);
+  });
+  it('SLANDLES can parse a manifest containing a particle with an argument list', async () => {
+    const manifest = await ManifestPrimitiveParser.parse(`
+    particle TestParticle in 'a.js'
+      list: reads [Product {}]
+      person: writes Person {}
+      thing: \`consumes Slot
+        otherThing: \`provides Slot
+    `);
+    const particles = ManifestPrimitiveParser.extract('particle', manifest);
+    assert.lengthOf(particles, 1);
+  });
+  it('can parse a manifest with dependent handles', async () => {
+    const manifest = await ManifestPrimitiveParser.parse(`
+    particle TestParticle in 'a.js'
+      input: reads [Product {}]
+        output: writes [Product {}]
+      thing: consumes
+        otherThing: provides
+    `);
+    const particles = ManifestPrimitiveParser.extract('particle', manifest);
+    assert.lengthOf(particles, 1);
+  });
+  it('SLANDLES can parse a manifest with dependent handles', async () => {
+    const manifest = await ManifestPrimitiveParser.parse(`
+    particle TestParticle in 'a.js'
+      input: reads [Product {}]
+        output: writes [Product {}]
+      thing: \`consumes Slot
+        otherThing: \`provides Slot
+    `);
+    const particles = ManifestPrimitiveParser.extract('particle', manifest);
+    assert.lengthOf(particles, 1);
+  });
+  // TODO(sjmiles): redundant? previous tests contain schema
+  it('can parse a manifest containing a schema', async () => {
+    const manifest = await ManifestPrimitiveParser.parse(`
+      schema Bar
+        value: Text
+    `);
+    const schema = ManifestPrimitiveParser.extract('schema', manifest)[0] as AstNode.Schema;
+    assert(schema, 'failed to parse any schema');
+    const item: AstNode.SchemaItem = schema.items[0];
+    assert.strictEqual(item.kind, 'schema-field', 'schema-field type expected as first item');
+    const field: AstNode.SchemaField = item as AstNode.SchemaField;
+    assert.strictEqual(field.name, 'value', 'field has unexpected name');
+    assert.strictEqual(field.type.kind, 'schema-primitive', 'field expected to be schema-primitive type');
+    const type = field.type as AstNode.SchemaPrimitiveType;
+    assert.strictEqual(type.type, 'Text', 'field has unexpected type');
+  });
+  it('can parse a manifest containing an inline schema with line breaks and a trailing comma', async () => {
+    const manifest = await ManifestPrimitiveParser.parse(`
+      particle Fooer
+        foo: reads Foo {
+          // Comments can go here
+          value: Text,
+          other: Number, // Or here.
+        }
+    `);
+    // passes if the parser doesn't throw
```

Comment:
Maybe just assert some basic property on the output?

`assert.include(manifest[0] as AstNode.Particle, {kind: 'particle', name: 'Fooer'});`

Reply:
Done